### PR TITLE
[DONE] Merge accounts

### PR DESCRIPTION
**Pull request by chabanenk0**
- listener (done)
- method for merging accounts (rewrite all the connections of current users to another one, done)
- improve user's message when emails are the same (done)
- modify user's dependencies with update queries (done)

### CODE REVIEW

File: src/Geekhub/UserBundle/EventListener/AccountsMergeSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+            $email = $object->getEmail();
+            $em = $this->container->get('doctrine')->getManager();
+            $userWithTheSameEmail = $em->getRepository('GeekhubUserBundle:User')->findOneByEmail($email);
+
+            if ($userWithTheSameEmail && ($userWithTheSameEmail->getId() != $object->getId())) {
+                $mergeRequest =  new MergeRequest();
+                $mergeRequest->setProposersId($object->getId());
+                $mergeRequest->setMergingUserId($userWithTheSameEmail->getId());
+                $hash = $this->getUniqueHash();
```
Comment (spolischook) [inline src/Geekhub/UserBundle/EventListener/AccountsMergeSubscriber.php:64]: Use this function uniqid - http://www.php.net/manual/en/function.uniqid.php

File: src/Geekhub/UserBundle/EventListener/AccountsMergeSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        $body = $this->container->get('templating')->render(
+            'GeekhubResourceBundle:Email:accounts_merge_notification.html.twig',
+                array(
+                    'user' => $userWithTheSameEmail->getFirstName()." ".$userWithTheSameEmail->getLastName(),
+                    'userWhoAsksMerge' => $userWhoAsksMerge->getFirstName()." ".$userWhoAsksMerge->getLastName(),
+                    'hash' => $hash,
+                )
+            );
+
+        $message->setFromEmail('test@gmail.com')
```
Comment (spolischook) [inline src/Geekhub/UserBundle/EventListener/AccountsMergeSubscriber.php]: Use parameters for sender mail

File: src/Geekhub/UserBundle/EventListener/AccountsMergeSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            ->setFromName('Черкаська мрія')
```
Comment (spolischook) [inline src/Geekhub/UserBundle/EventListener/AccountsMergeSubscriber.php]: Use parameters for sender name

File: src/Geekhub/UserBundle/Entity/MergeRequest.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    /**
+     * @param mixed $proposer
+     */
+    public function setProposersId($proposersId)
+    {
+        $this->proposersId = $proposersId;
+    }
+}
```
Comment (ahonymous) [inline src/Geekhub/UserBundle/Entity/MergeRequest.php:105]: Why you created more entities?

Comment (chabanenk0) [inline src/Geekhub/UserBundle/Entity/MergeRequest.php:105]: This entity (table) collects user's requests for account's merge. We need to store it before the user conforms a merge. Any other ideas?

Comment (spolischook) [inline src/Geekhub/UserBundle/Entity/MergeRequest.php:105]: +1 to  Any other ideas.

Comment (ahonymous) [inline src/Geekhub/UserBundle/Entity/MergeRequest.php:105]: 1) user written data to inputs and send to action updateContactsAction
2) in UserController you check request data and if other user has this email, you save flashMessage (This email is busy!!! Please added other account social networks in you profile or write other email) and return updateContacts form with flashMessage

Comment (spolischook) [inline src/Geekhub/UserBundle/Entity/MergeRequest.php:105]: good, but the ticket about merge accaunts

**Pull request closed without merging**